README.md

```diff
@@ -1,10 +1,75 @@
+what is set as the current device type and the current artifact name. The
+command line options of current device and current artifact are used for
+building the _update request to the server_, and can hence specify dummy names,
+so that one can mismatch on the Artifact name, and match on the client type. In
+general these flags should match up though, and hence be the same. If not
+specified these will default to *test*.
+
+* As mentioned above, in general the client has **no** state. However, this is
+not wholly true, as the client will generate missing keys, and then reuse these
+keys on second startup. If the client has no keys to start out with, it will
+generate the number of keys needed, one for each client, and store them in the
+`keys/` folder. Thus on killing the clients, and starting them back up, the keys
+will still be present in the directory, and the clients will start back up with
+the same keys it had on the previous run. If it needs to start more clients than
+there are already keys present, it will start up one client for each key, and
+then go at generating a new one for each client which is missing one.
+
+* As mentioned, the client has no state! This implies that it can be 'updated'
+with the same artifact N times, and still report the same inventory it had at
+startup. This is because an update is always discarded on the client side (it is
+written to /dev/null, and not parsed by mender-artifact).
+
+* It reports the update phases, _downloading_, _installing_, _rebooting_ and
+_success_ or _failure_. The time the client spends in between each of these
+stages is determined by the CLI-parameter `-wait`, and defaults to `1800`. This
+does not mean that each device waits for `1800` seconds between each update
+stage though. This is the maximum time that the client will wait in between each
+update step on the update path (i.e., downloading, installing, rebooting), where
+the wait is some amount of seconds on the uniform interval `[0,wait)`.
+
+
+## Working with the Demo Server
+
+Following are the considerations needed when running the client with the Mender
+Demo server. In order to run the Demo server with *N* clients, (this assumes the
+demo server is already running), there are a few options that can be worthwhile
+consideration.
+
+* First the number of clients is specified with the `-count` flag. If this is a
+  fairly large number, consider using the `-interval=<last-start>` flag, in
+  order to not bombard the server with N authorization calls at once. This can
+  be used to bring the clients up at an interval from `[time-now,time-now +
+  last-start]`. Hence, significantly lightening the upstart load on the server.
+
+* Also consider how many of the devices should fail during a simulated update.
+This can be controlled with the `-failcount` flag, and will default to one, if
+not specified.
+
+* The address of the Demo server is assumed by default to be on `localhost`, but
+can be overridden by the `-backend` flag.
```

Comment:
General comment: even though we use a single `-`, these are really parameters (i.e. expect a value) rather than flags. It is repeated everywhere and I think we should change it. Even better if we indicate a value is expected, for example `-backend=<server-url>` (as done above for `-interval`).

Reply:
Agreed. Fixing. But you still prefer single `-`, over `--` ?

Reply:
It must be simple as that is the way it is implemented and showed in the help text (I know that internally the implementation accept double `--`, but we should not count on that).